app/javascript/components/enrollment/steps/Identification.js

```diff
@@ -22,7 +25,11 @@ class Identification extends React.Component {
       current: { ...this.props.currentState },
       errors: {},
       modified: {},
-      languageOptions: this.getLanguageOptions(),
+      primaryLanguageMessage: this.getPrimaryLanguageSupportMessage(tryToMatchLanguage(this.props.currentState.patient.primary_language)),
+      // The dropdown for languages requires the selected values to be in a special format
+      // maintain these values in state to avoid unnecessary computation in `render`
+      selectedPrimaryLanguage: this.formatLanguageAsOption(tryToMatchLanguage(this.props.currentState.patient.primary_language)),
```

Comment:
Is there a strong reason why we cannot just update `primary_language` and `secondary_language` on the patient (current and modified)? seems a bit redundant to have both

Reply:
It is possible to update the values on `patient` directly, but I disagree with doing so.
1. The value on `Patient` would need to be in the format accepted by our React-Select component. Seems silly to modify all the front-end code just for slightly-easier code in one file. (`primary_language` would become `{label: 'English', value: 'eng'}`).
2. Maintaining two extra variables in state is not expensive or bad/ugly code.
3. The documentation clearly explains why we are doing this, so it shouldn't be confusing.
4. Larger changes to the front-end could introduce bugs. The KISS principle as always, applies.

Reply:
I've updated the way this is handled, so this whole problem is kinda moot.